projects/storefrontlib/src/cms-components/product/product-list/container/product-list-component.service.ts

```diff
@@ -47,13 +52,32 @@ export class ProductListComponentService {
     protected currencyService: CurrencyService,
     protected languageService: LanguageService,
     protected router: Router
-  ) {}
+  ) {
+    this.router.events
+      .pipe(filter((e: RouterEvent): e is Scroll => e instanceof Scroll))
+      .subscribe(e => {
+        if (e.position) {
+          this.autoScrollPosition = e.position;
```

Comment:
one question, if we scroll all the way to the bottom http://localhost:4200/electronics-spa/en/USD/Open-Catalogue/Cameras/Digital-Cameras/c/575 and click the product, it will not bring you at the position you were in before.

Reply:
nvm, what I said above, because it depends on how fast it gets loaded, if I throttle the network, sometimes its above 300ms with the logic you added, but I think we can make it better (i'll put another comment on that statement)